Reconstruction/RecCalorimeter/src/components/NoiseCaloCellsFromFileTool.cpp

```diff
@@ -0,0 +1,206 @@
+  //Delete histograms & file
+  debug() << "Deleting histograms with noise constants" << endmsg;
+  for (auto it: m_histoElecNoiseConst) {
+    delete it;
+  }
+  if (m_addPileup) {
+    for (auto it: m_histoPileupConst) {
+      delete it;
+    }
+  }
+  delete m_file;
+
+  StatusCode sc = GaudiTool::finalize();
+  return sc;
+
+}
+
+StatusCode NoiseCaloCellsFromFileTool::initNoiseFromFile() {
+  // check if file exists
+  if ( m_noiseFileName.empty()) {
+    error() << "Name of the file with noise values not set" << endmsg;
+    return StatusCode::FAILURE;
+  }
+  m_file = new TFile(m_noiseFileName.c_str(),"READ");
+  if (m_file->IsZombie()) {
+    error() << "Couldn't open the file with noise constants" << endmsg;
+    return StatusCode::FAILURE;
+  }
+  else {
+    info() << "Opening the file with noise constants: "<<m_noiseFileName << endmsg;
+  }
+
+  std::string elecNoiseLayerHistoName, pileupLayerHistoName;
+  // Read the histograms with electronics noise and pileup from the file
+  for (unsigned i = 0; i<m_numRadialLayers; i++) {
+
+    elecNoiseLayerHistoName = m_elecNoiseHistoName+std::to_string(i+1);
+    debug()<<"Getting histogram with a name "<<elecNoiseLayerHistoName<<endmsg;
+    m_histoElecNoiseConst.push_back(dynamic_cast<TH1F*>(m_file->Get(elecNoiseLayerHistoName.c_str())));
+    if (m_histoElecNoiseConst.at(i)==nullptr) {
+      error()<<"Histogram  "<<elecNoiseLayerHistoName<<" does not exist!!!!"<<endmsg;
+      return StatusCode::FAILURE;
+    }
+    if (m_addPileup) {
+      pileupLayerHistoName = m_pileupHistoName+std::to_string(i+1);
+      debug()<<"Getting histogram with a name "<<pileupLayerHistoName<<endmsg;
+      m_histoPileupConst.push_back(dynamic_cast<TH1F*>(m_file->Get(pileupLayerHistoName.c_str())));
+      if (m_histoPileupConst.at(i)==nullptr) {
+	error()<<"Histogram  "<<pileupLayerHistoName<<" does not exist!!!!"<<endmsg;
+	return StatusCode::FAILURE;
+      }
+    }
+  }
+
+  //Check if we have same number of histograms (all layers) for pileup and electronics noise
+  if (m_histoElecNoiseConst.size()==0) {
+    error()<<"No histograms with noise found!!!!"<<endmsg;
+    return StatusCode::FAILURE;
+  }
+  if (m_addPileup) {
+    if (m_histoElecNoiseConst.size()!=m_histoPileupConst.size()) {
+      error()<<"Missing histograms! Different number of histograms for electronics noise and pileup!!!!"<<endmsg;
+      return StatusCode::FAILURE;
+    }
+  }
+
+  return StatusCode::SUCCESS;
+}
+
+double NoiseCaloCellsFromFileTool::getNoiseConstantPerCell(int64_t aCellId) {
+
+  double elecNoise = 0.;
+  double pileupNoise = 0.;
+
+  //Get cell coordinates: eta and radial layer
+  double cellEta = m_segmentation->eta(aCellId);
+  // Take readout, bitfield from GeoSvc
+  auto decoder = m_geoSvc->lcdd()->readout(m_readoutName).idSpec().decoder();
+  decoder->setValue(aCellId);
+  unsigned cellLayer = (*decoder)[m_activeFieldName];
+
+  //double cellR = det::utils::tubeDimensions(aCellId).x();
```

Comment:
Please remove code that is not needed anymore. If you want to keep it as a reference add a comment above in which circumstance this may be needed / why it is not needed at the moment.